js/src/components/contact-information/phone-number-card.js

```diff
@@ -0,0 +1,144 @@
+import Section from '.~/wcdl/section';
+import SelectControl from '.~/wcdl/select-control';
+import AppInputControl from '.~/components/app-input-control';
+import AccountCard, { APPEARANCE } from '.~/components/account-card';
+import AppButton from '.~/components/app-button';
+import AppSpinner from '.~/components/app-spinner';
+import './phone-number-card.scss';
+
+const noop = () => {};
+
+function PhoneNumberContent( {
+	initCountry,
+	initNationalNumber,
+	onPhoneNumberChange,
+} ) {
+	const countryCallingCodes = useCountryCallingCodes( 'select-options' );
+	const [ country, setCountry ] = useState( initCountry );
+	const [ number, setNumber ] = useState( initNationalNumber );
+
+	const handleChange = ( nextCountry, nextNumber ) => {
+		if ( nextCountry !== country ) {
+			setCountry( nextCountry );
+		}
+		if ( nextNumber !== number ) {
+			setNumber( nextNumber );
+		}
+
+		const countryCallingCode = nextCountry
+			? getCountryCallingCode( nextCountry )
+			: '';
+		onPhoneNumberChange( countryCallingCode, nextNumber, nextCountry );
+	};
+
+	const handleCountryChange = ( v ) => handleChange( v, number );
+	const handleNumberChange = ( v ) => handleChange( country, v );
+
+	return (
+		<Section.Card.Body>
+			<Flex gap={ 4 }>
+				<FlexItem>
+					<SelectControl
+						label={ __(
+							'Country code',
+							'google-listings-and-ads'
+						) }
+						isSearchable
+						excludeSelectedOptions={ false }
+						options={ countryCallingCodes }
+						selected={ country }
+						onChange={ handleCountryChange }
+					/>
+				</FlexItem>
+				<FlexBlock>
+					<AppInputControl
+						label={ __(
+							'Phone number',
+							'google-listings-and-ads'
+						) }
+						value={ number }
+						onChange={ handleNumberChange }
+					/>
+				</FlexBlock>
+			</Flex>
+		</Section.Card.Body>
+	);
+}
+
+export default function PhoneNumberCard( {
+	isPreview,
+	initEditing,
+	onEditClick,
+	onPhoneNumberChange = noop,
+} ) {
+	const [ isEditing, setEditing ] = useState( initEditing );
+	const { loaded, data } = useGoogleMCPhoneNumber();
+
+	const editButton = isEditing ? null : (
+		<AppButton
+			isSecondary
+			onClick={ () => {
+				if ( onEditClick ) {
+					onEditClick();
+				} else {
+					setEditing( true );
+				}
+			} }
+		>
+			{ __( 'Edit', 'google-listings-and-ads' ) }
+		</AppButton>
+	);
+
+	let description = null;
+	let phoneNumberContent = null;
+
+	if ( isEditing ) {
+		description = __(
+			'Please enter a phone number to be used for verification.',
+			'google-listings-and-ads'
+		);
+
+		if ( loaded ) {
+			phoneNumberContent = (
+				<>
+					<CardDivider />
+					<PhoneNumberContent
+						initCountry={ data.country }
+						initNationalNumber={ data.nationalNumber }
+						onPhoneNumberChange={ onPhoneNumberChange }
+					/>
+				</>
+			);
+		} else {
+			phoneNumberContent = <AppSpinner />;
+		}
+	} else {
+		description = loaded ? data.display : <Spinner />;
```

Comment:
:question:  Having the preloading logic here and in https://github.com/woocommerce/google-listings-and-ads/pull/917/files#diff-640c9e5414f85abfa4da47b6a020028e033ba15fce7aac4ae0a975a4efe49c86R115-R128 looks like too much.
I think we can make containing element render this component. Then, this component could decide whether to render a scaffold with a tiny preloader in place of a phone number, or a fully prepared input control.

Also, in one case we use `<Spinner>` in the other `<AppSpinner>`. Is that on purpose?

Reply:
Sorry, I'm not quite sure of your meaning. Could you provide some code examples?

The logic is used for rendering different visual results.

![Kapture 2021-08-03 at 11 36 33](https://user-images.githubusercontent.com/17420811/127955340-9827b893-bf42-4fef-b8d2-5cc29445223e.gif)

Reply:
Sorry, I put the wrong link.
My main point was about handling loaded here and in `ContactInformation`
https://github.com/woocommerce/google-listings-and-ads/pull/917/files#diff-51ff763be0b20871d193755311c5c437caafd6c4d46bb1413de599e1124a7714R53
what results in 
![image](https://user-images.githubusercontent.com/17435/127996658-bf259ca4-7721-421e-9fe5-11262de8ec5f.png)

Here is code sample/suggestion https://github.com/woocommerce/google-listings-and-ads/pull/917/files#r681624815

Reply:
Speaking of `<Spinner>` vs. `<AppSpinner>` now I got it, thanks. I agree we need both here. I got mislead by their names. I thought `App-` one is just profiled for our need, while they are in fact `<InlineSpinenr>` and `<BlockSpinner>`.

Reply:
:nail_care: :scroll: For the block spinner `<AppSpinner>`, I'd adapt it's position/margin/padding to make the card fixed height = do not jump once loaded.

Reply:
I changed the loading UI in the later PR #926, and it shows a spinner with a card layout instead. The demo in that PR:

https://user-images.githubusercontent.com/17420811/127852973-dcbed8d8-f2ea-4dc5-91f4-85e66199617d.mp4